src/lib/storage/WebFileSystemStorage.ts

```diff
@@ -67,6 +70,29 @@ export class WebFileSystemStorage implements FileStorage {
     await writable.close();
   }
 
+  async appendFile(uri: string, data: string | Blob): Promise<void> {
+    const filepath = fileUriToPath(uri);
+    if (!pathIsFile(filepath)) {
+      throw new Error(`Cannot append to a directory: ${uri}`);
+    }
+    const dirname = getDirname(filepath);
+    const filename = cast(getFilename(filepath));
+
+    await this.globalAppendLock.acquire();
```

Comment:
Maybe use web locks (navigator.locks.request) instead? That would block writers in different tabs.